lib/pymedphys/_dicom/connect/send.py

```diff
@@ -0,0 +1,117 @@
+# Copyright (C) 2020 University of New South Wales & Ingham Institute
+# Copyright (C) 2020 Stuart Swerdloff and Simon Biggs
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+
+from pymedphys._imports import pydicom, pynetdicom
+
+from pymedphys._dicom.connect.base import DicomConnectBase
+
+
+class DicomSender(DicomConnectBase):
+    """Class which provides SCU functionality to send DICOM objects"""
+
+    def verify(self):
+        """Verify that we can connect to the DICOM endpoint using C_ECHO
+
+        Returns
+        -------
+        bool
+            True if the C_ECHO was successful, False otherwise.
+        """
+
+        ae = pynetdicom.AE(ae_title=self.ae_title)
+        ae.requested_contexts = pynetdicom.VerificationPresentationContexts
+
+        # Associate with a peer DICOM AE
+        assoc = ae.associate(self.host, self.port, ae_title=self.ae_title)
+
+        result = None
+
+        if assoc.is_established:
+            status = assoc.send_c_echo()
+
+            if status:
+                result = status.Status
+
+            # Release the association
+            assoc.release()
+
+        return not result is None
+
+    def send(self, dcm_files):
+        """Send each DICOM object to the configured DICOM location
+
+        Parameters
+        ----------
+        dcm_files : list
+            list of str, pathlib.Path or pydicom.Dataset to send
+
+        Returns
+        -------
+        list
+            List containing status objects for each DICOM file sent
+        """
+
+        transfer_syntax = [pydicom.uid.ImplicitVRLittleEndian]
+
+        ae = pynetdicom.AE()
+
+        for context in pynetdicom.StoragePresentationContexts:
+            ae.add_requested_context(context.abstract_syntax, transfer_syntax)
+
+        assoc = ae.associate(self.host, self.port, ae_title=self.ae_title)
+
+        statuses = []
+        if assoc.is_established:
+
+            for dataset in dcm_files:
+
+                if not isinstance(dataset, pydicom.Dataset):
+
+                    try:
+                        dataset = pydicom.read_file(dataset)
+                    except pydicom.errors.InvalidDicomError as ide:
+                        logging.warning(ide)
+                        logging.warning(
+                            "Unable to read DICOM file, skipping: %s", dataset
```

Comment:
Could we opt for the fail fast philosophy here? Instead have this try except block happen at the highest level function (the CLI), and have this function just be in charge of sending the DICOM files it is provided. If those files are invalid, I think having the `send` method raise an error is appropriate.